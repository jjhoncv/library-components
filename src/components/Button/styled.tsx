import styled, { keyframes } from 'styled-components';
import { COLORS, FONTS } from './../../ui'

interface Props {
  fullWidth?: boolean;
  disabled?: boolean;
  loading?: boolean;
  primary?: boolean;
}

const animationLoading = keyframes`
  to {transform: rotate(360deg);}
`;

export const ButtonLoading = styled.div`
  display: inline-block;
  box-sizing: border-box;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: 2px solid ${COLORS.BLACK};
  border-top: 2px solid ${COLORS.LIGHT_GRAY};
  animation: ${animationLoading} .6s linear infinite;

  ${(props: Props) => props.primary && `
    border: 2px solid ${COLORS.WHITE};
    border-top: 2px solid ${COLORS.BLACK};
  `};
`;

export const ButtonStyled = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${(props: Props) => props.fullWidth ? '100%' : 'auto'};
  background-color: ${COLORS.LIGHT_GRAY};
  height: 44px;
  padding: 12px 25px;
  border: none;
  border: 1px solid ${COLORS.OPAQUE_GRAY};
  border-radius: 4px;
  color: ${COLORS.BLACK};
  font-size: 14px;
  font-family: ${FONTS.ROBOTO_REGULAR};
  cursor: pointer;

  &:hover {
    background-color: ${COLORS.HOVER_GRAY};
  }

  ${(props: Props) => props.disabled && `
    pointer-events: none;
    user-select: none;
    opacity: 0.3;
  `};

  ${(props: Props) => props.loading && `
    pointer-events: none;
    user-select: none;
  `};

  ${(props: Props) => props.primary && `
    background:${COLORS.BLACK};
    color: ${COLORS.WHITE};

    &:hover {
      background-color: ${COLORS.HOVER_BLACK};
    }
  `};
`;
