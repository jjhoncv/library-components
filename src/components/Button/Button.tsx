import * as React from 'react';
import * as PropTypes from 'prop-types';
import { ButtonStyled, ButtonLoading } from './styled';

interface Props {
  className?: string;
  primary?: boolean;
  disabled?: boolean;
  onClick?: Function;
  children?: any;
  loading?: boolean;
  fullWidth?: boolean;
  dataTest?: string;
}

export const Button: React.FunctionComponent<Props> = ({
  children,
  loading,
  ...nextProps
}) => {
  return (
    <ButtonStyled
      loading={loading}
      {...nextProps}
    >
      { loading ? <ButtonLoading {...nextProps} /> : children }
    </ButtonStyled>
  );
} ;

Button.propTypes = {
  className: PropTypes.string,
  primary: PropTypes.bool,
  children: PropTypes.any,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  onClick: PropTypes.func,
  dataTest: PropTypes.string,
  fullWidth: PropTypes.bool
}
