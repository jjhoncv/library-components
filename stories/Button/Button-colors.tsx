import * as React from 'react';
import styled from 'styled-components'
import { WithPropTypes } from '../../.storybook/Containers/WithPropTypes';
import { Button } from '../../src/components/Button';

const ButtonWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 50%;
  height: 200px;
  justify-content: space-around;
  align-items: center;
`;

const Wrapper = () => (
  <ButtonWrapper>
    <Button
      onClick={()=> {console.log("Le diste click")}}
    >
      Default
    </Button>
    <Button
      primary
      onClick={()=> {console.log("Le diste click")}}
    >
      Primary
    </Button>
  </ButtonWrapper>
);

export default () => {
  return WithPropTypes({
    Component: Button,
    RenderComponent: Wrapper,
    showProp: false
  });
}
