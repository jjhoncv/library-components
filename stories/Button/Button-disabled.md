# Button

Importaci√≥n del componente Button.

```jsx
import { Button } from 'library/components/Button';
```

<!-- STORY -->

#### Uso

```jsx
<Button
  onClick={()=> {console.log("Le diste click")}}
  disabled
>
  Default
</Button>
<Button
  primary
  onClick={()=> {console.log("Le diste click")}}
  disabled
>
  Primary
</Button>
```
