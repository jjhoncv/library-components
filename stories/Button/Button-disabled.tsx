import * as React from 'react';
import styled from 'styled-components'
import { WithPropTypes } from '../../.storybook/Containers/WithPropTypes';
import { Button } from '../../src/components/Button/Button';

const ButtonWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 50%;
  height: 200px;
  justify-content: space-around;
  align-items: center;
`;

const Wrapper = () => (
  <ButtonWrapper>
    <Button
      onClick={()=> {console.log("Le diste click")}}
      disabled
    >
      Default
    </Button>
    <Button
      primary
      onClick={()=> {console.log("Le diste click")}}
      disabled
    >
      Primary
    </Button>
  </ButtonWrapper>
);

export default () => {
  return WithPropTypes({
    Component: Button,
    RenderComponent: Wrapper,
    showProp: false
  });
}
