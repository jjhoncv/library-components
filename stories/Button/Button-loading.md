# Button

Importaci√≥n del componente Button.

```jsx
import { Button } from 'library/components/Button';
```

<!-- STORY -->

#### Uso

```jsx
<Button
  onClick={()=> {console.log("Le diste click")}}
  loading
>
  Default
</Button>
<Button
  primary
  onClick={()=> {console.log("Le diste click")}}
  loading
>
  Primary
</Button>
```
