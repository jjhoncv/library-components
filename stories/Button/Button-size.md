# Button

Importaci√≥n del componente Button.

```jsx
import { Button } from 'library/components/Button';
```

<!-- STORY -->

#### Uso

```jsx
<Button
  onClick={()=> {console.log("Le diste click")}}
>
  Default
</Button>
<Button
  onClick={()=> {console.log("Le diste click")}}
  fullWidth
>
  Default
</Button>
<Button
  primary
  onClick={()=> {console.log("Le diste click")}}
>
  Primary
</Button>
<Button
  primary
  onClick={()=> {console.log("Le diste click")}}
  fullWidth
>
  Primary
</Button>
```
