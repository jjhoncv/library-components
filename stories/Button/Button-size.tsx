import * as React from 'react';
import styled from 'styled-components'
import { WithPropTypes } from '../../.storybook/Containers/WithPropTypes';
import { Button } from '../../src/components/Button/Button';

const ButtonWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  height: 200px;
  align-items: center;
`;

const Wrapper = () => (
  <ButtonWrapper>
    <Button
      onClick={()=> {console.log("Le diste click")}}
    >
      Default
    </Button>
    <Button
      onClick={()=> {console.log("Le diste click")}}
      fullWidth
    >
      Default
    </Button>
    <Button
      primary
      onClick={()=> {console.log("Le diste click")}}
    >
      Primary
    </Button>
    <Button
      primary
      onClick={()=> {console.log("Le diste click")}}
      fullWidth
    >
      Primary
    </Button>
  </ButtonWrapper>
);

export default () => {
  return WithPropTypes({
    Component: Button,
    RenderComponent: Wrapper,
    showProp: false
  });
}
