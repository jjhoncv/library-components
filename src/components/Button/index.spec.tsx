import * as React from 'react';
import { shallow } from 'enzyme';
import { Button } from './Button';

let propsButton = {
  className:'c-button',
  onClick: ()=>{},
}

describe('Button Component', () => {
  describe('Button => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <Button {...propsButton}>
          Test
        </Button>
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('Button Builder', () => {
    it('Button => render text', () => {
      const wrapper = shallow(<Button>Test</Button>);
      expect(wrapper.prop('children')).toBe('Test');
    })
    it('Button => render with prop disabled', () => {
      const props = {
        className: 'c-button',
        disabled: true
      }
      const wrapper = shallow(<Button {...props}>Test</Button>);
      expect(wrapper.prop('disabled')).toBe(true);
    })
    it('Button => render with prop loading', () => {
      const props = {
        className: 'c-button',
        loading: true
      }
      const wrapper = shallow(<Button {...props}>Test</Button>);
      expect(wrapper.prop('loading')).toBe(true);
    })
  });
});


