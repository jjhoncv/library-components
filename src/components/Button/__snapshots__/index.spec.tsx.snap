// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button => render snapshot Render correctly 1`] = `"<button class=\\"c-button sc-bwzfXH bPIpeZ\\">Test</button>"`;
